### PR TITLE
[4.0] Setup & Page Creation updates

### PR DESCRIPTION
**Pull request by weaverryan**
Hi guys!

I'm currently re-reading the most important changes and updating for 4.0 (probably, we need to do a scripted find-replace for the biggest changes (e.g. `AppBundle` -> `App`, `app/config/config.yml` to... other things) for the bulk of the chapters.

I'm already continuing with routing and other chapters, but wanted to get this PR at least submitted as soon as possible.

### CODE REVIEW

File: page_creation.rst [hunk 1]
```diff
@@ -65,44 +60,154 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+Then, in ``config/routes.yaml``, remove the route we just created and uncomment
+the annotation route import at the bottom:
+
+.. code-block:: yaml
+
+    controllers:
+        resource: ../src/Controller/
+        type: annotation
+
+After this one-time setup, you can nowadd your route directly *above* the controller:
```
Comment (javiereguiluz) [inline page_creation.rst]: `nowadd` -> `now add`

File: page_creation.rst [hunk 2]
```diff
@@ -65,44 +60,154 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+You may not have noticed, but when you ran ``composer require annotations``, two
+special things happened, both thanks to a powerful Composer plugin called
+:doc:`Flex </setup/flex>`.
+
+First, ``annotations`` isn't a real package name: it's an *alias* (i.e. shortcut)
+that Flex resolves to ``sensio/framework-extra-bundle``.
+
+Second, after this package was downloaded, Flex executed a *recipe*, which automatically
+enabled the bundle. Flex recipes exist for many packages (not just bundles) and have
+have the ability to do a lot, like adding configuration files, creating directories,
```
Comment (javiereguiluz) [inline page_creation.rst]: `have` is repeated at the end of the previous line.

File: page_creation.rst [hunk 3]
```diff
@@ -134,13 +239,13 @@ variable so we can render that::
         }
     }
 
-Finally, template files should live in the ``app/Resources/views`` directory. Create
-a new ``app/Resources/views/lucky`` directory with a new ``number.html.twig`` file
-inside:
+Template files live in the ``templates/`` directory, which was created for your automatically
```
Comment (javiereguiluz) [inline page_creation.rst]: `for your` -> `for you`

File: page_creation.rst [hunk 4]
```diff
@@ -155,17 +260,17 @@ other templates and leverage its powerful layout inheritance system.
...
-Great news! You've already worked inside the two most important directories in your
+Great news! You've already worked inside the most important directories in your
 project:
 
-``app/``
-    Contains things like configuration and templates. Basically, anything
-    that is *not* PHP code goes here.
+``config/``
+    Contains... configuration of course!. You will configure routes, :doc:`services </service_container>`
+    and pckages.
```
Comment (javiereguiluz) [inline page_creation.rst]: `pckages` -> `packages`

File: page_creation.rst [hunk 5]
```diff
@@ -174,58 +279,17 @@ So what about the other directories in the project?
     The famous ``bin/console`` file lives here (and other, less important
     executable files).
 
-``tests/``
-    The automated tests (e.g. Unit tests) for your application live here.
-
 ``var/``
     This is where automatically-created files are stored, like cache files
-    (``var/cache/``), logs (``var/logs/``) and sessions (``var/sessions/``).
+    (``var/cache/``) and logs (``var/logs/``).
```
Comment (javiereguiluz) [inline page_creation.rst]: Be careful!! `var/logs/` is now called `var/log/` See https://github.com/symfony/recipes/pull/169

File: setup.rst [hunk 6]
```diff
@@ -4,135 +4,111 @@
...
-a new Symfony application as explained in the previous sections. Instead,
-someone else has already created and submitted it to a shared repository.
+Setting up an Existing Symfony Project
+--------------------------------------
 
-It's recommended to not submit some files (``.env``) and directories (``vendor/``,
-cache, logs) to the repository, so you'll have to do the following when
-installing an existing Symfony application:
+If you're working on an existing Symfony application, you'll just need to do a few
+things to get your project setup. Assuming your team uses git, you can setup your
```
Comment (javiereguiluz) [inline setup.rst]: `git` -> `Git`

**Review by javiereguiluz — APPROVED** [review]
Nice changes! Thanks for this.

File: page_creation.rst [hunk 7]
```diff
@@ -65,44 +60,154 @@ random) number and prints it. To do that, create a "Controller class" and a
         }
     }
 
-Before diving into this, test it out! If you are using PHP's internal web server
-go to:
+To map a URL to this controller, create a route in ``config/routes.yaml``:
```
Comment (javiereguiluz) [inline page_creation.rst]: I just wonder if this is too advanced to be understood by someone who is starting with Symfony:

```
To map a URL to this controller, create a route in ``config/routes.yaml``:
```

Would something like this be better ... or just more verbose and complicated?

```
Now you need to associate this controller with a public URL so the ``number()``
method is executed when users browse some URL (e.g. ``/lucky/number``). This
association is defined creating a **route** in the ``config/routes.yaml`` file:
```

File: page_creation.rst [hunk 8]
```diff
@@ -65,44 +60,154 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+-----------------------------------------
+
+You may not have noticed, but when you ran ``composer require annotations``, two
+special things happened, both thanks to a powerful Composer plugin called
+:doc:`Flex </setup/flex>`.
+
+First, ``annotations`` isn't a real package name: it's an *alias* (i.e. shortcut)
+that Flex resolves to ``sensio/framework-extra-bundle``.
+
+Second, after this package was downloaded, Flex executed a *recipe*, which automatically
```
Comment (javiereguiluz) [inline page_creation.rst]: In this paragraph I'm missing a simple description of what a recipe is. If you agree, I propose something like: *"a set of automated instructions that tell Symfony how to integrate some external package"*. So this is how it'd look:

```
Second, after this package was downloaded, Flex executed a *recipe*, which is a
set of automated instructions that tell Symfony how to integrate some external
package. Flex recipes exist for many packages (not just bundles) and have the
ability to do a lot, like adding configuration files, creating directories,
updating ``.gitignore`` and adding new config to your ``.env`` file. Flex
*automates* the installation of packages so you can get back to coding.
```

File: page_creation.rst (hunk 5 above)
Comment (weaverryan) [reply page_creation.rst]: Oh! I hadn't even realized! Thanks :)

File: configuration.rst [hunk 9]
```diff
@@ -295,6 +295,15 @@ signs - e.g. ``%locale%``.
...
 
+.. _config-dot-env:
+
+The .env File
+~~~~~~~~~~~~~
+
+There is also a ``.env`` file which is loaded. Its contents become environment variables
+in the dev environment, making it easier to reference environment variables in your
+code.
+
```
Comment (weaverryan) [inline configuration.rst]: This will be covered more #8546. I just bootstrapped this section so that I could link to it.

**Comment by weaverryan** [comment]
All comments addressed!

File: page_creation.rst [hunk 10]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+You may not have noticed, but when you ran ``composer require annotations``, two
+special things happened, both thanks to a powerful Composer plugin called
+:doc:`Flex </setup/flex>`.
+
+First, ``annotations`` isn't a real package name: it's an *alias* (i.e. shortcut)
+that Flex resolves to ``sensio/framework-extra-bundle``.
+
+Second, after this package was downloaded, Flex executed a *recipe*, which is a
+set of automated instructions that tell Symfony how to integrate an external
+package. Flex recipes exist for many packages (not just bundles) and have the
```
Comment (javiereguiluz) [inline page_creation.rst]: One last comment on my side. In this paragraph we mention "bundles" for the first time and without prior introduction (*"[...] Flex recipes exist for many packages (not just bundles) and have [...]"*). Given that bundles are not important in this article, we could remove this mention. Thanks!

File: page_creation.rst (hunk 5 above)
Comment (yceruto) [reply page_creation.rst]: do you need some hand? see https://github.com/symfony/symfony-docs/pull/8547 ;)

File: page_creation.rst [hunk 11]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+-----------------
+
+Instead of defining your route in YAML, Symfony also allows you to use *annotation*
+routes. First, install the annotations package:
+
+.. code-block:: terminal
+
+    $ composer require annotations
+
+Then, in ``config/routes.yaml``, remove the route we just created and uncomment
```
Comment (wouterj) [inline page_creation.rst]: No first person in the docs

File: page_creation.rst [hunk 12]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+    controllers:
+        resource: ../src/Controller/
+        type: annotation
+
+After this one-time setup, you can now add your route directly *above* the controller:
+
+.. code-block:: diff
+
+    // src/Controller/LuckyController.php
+    // ...
```
Comment (wouterj) [inline page_creation.rst]: this and the empty line below should be swapped (minor)

File: page_creation.rst [hunk 13]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+
+Second, after this package was downloaded, Flex executed a *recipe*, which is a
+set of automated instructions that tell Symfony how to integrate an external
+package. Flex recipes exist for many packages and have the ability to do a lot, like
+adding configuration files, creating directories, updating ``.gitignore`` and adding
+new config to your ``.env`` file. Flex *automates* the installation of packages so
+you can get back to coding.
+
+You can learn more about Flex by reading ":doc:`/setup/flex`". But that's not necessary:
+Flex works automatically in the background when you add packages.
```
Comment (wouterj) [inline page_creation.rst]: This is probably the time to link to http://symfony.sh ?

File: page_creation.rst [hunk 14]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+
+The bin/console Utility
```
Comment (wouterj) [inline page_creation.rst]: The console utility?

File: page_creation.rst [hunk 15]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+-----------------------
+
+Your project already has a powerful debugging tool inside: the ``bin/console`` command.
+Try running it:
+
+.. code-block:: terminal
+
+    php bin/console
```
Comment (wouterj) [inline page_creation.rst]: Commands have to be prefixed by `$` (otherwise, the parser won't understand it)

File: page_creation.rst [hunk 16]
```diff
@@ -65,44 +60,157 @@ random) number and prints it. To do that, create a "Controller class" and a
...
+
+You should see a list of commands that can give you debugging information, help generate
+code, generate database migrations and a lot more. As you install more packages,
+you'll see more commands.
+
+To get a list of *all* of the routes in your system, use the ``debug:router`` command:
+
+.. code-block:: terminal
+
+    php bin/console debug:router
```
Comment (wouterj) [inline page_creation.rst]: same here

**Review by wouterj — CHANGES_REQUESTED** [review]
Mostly minor CS things

**Pull request closed without merging**